pkg/proxy/iptables/proxier.go

```diff
@@ -457,15 +467,15 @@ func (proxier *Proxier) OnServiceUpdate(allServices []api.Service) {
 				Port:           servicePort.Name,
 			}
 			activeServices[serviceName] = true
-			info, exists := proxier.serviceMap[serviceName]
-			if exists && proxier.sameConfig(info, service, servicePort) {
-				// Nothing changed.
-				continue
-			}
+			info, exists := newServiceMap[serviceName]
```

Comment:
It's weird to check newServiceMap to see if the service existed in oldServiceMap. I think the loop would make more sense if you started with an empty newServiceMap and copied pre-existing-unchanged services over from oldServiceMap, rather than starting with a complete copy of oldServiceMap and then deleting everything that had changed.